internal/pkg/externalplugins/labelblocker/labelblocker.go

```diff
@@ -115,35 +137,49 @@ func handle(cfg *externalplugins.Configuration, ctx labelCtx, gc githubClient, l
 
 		// If this rule does not match, try to match the next rule.
 		if !regex.MatchString(ctx.label) || !isMatchAction(ctx.action, blockLabel.Actions) {
-			log.Infof("%s:%s does not match regex or action", regex, ctx.action)
+			log.Infof("%s:%s does not match regex or action.", regex, ctx.action)
 			continue
 		}
 
 		// If the operator is a trusted user, don’t trigger blocking.
 		allTrustedUserLogins := listAllTrustedUserLogins(owner, blockLabel.TrustedTeams, blockLabel.TrustedUsers, gc, log)
 		if allTrustedUserLogins.Has(ctx.sender) {
-			log.Infof("Operator %s is trusted by the %s rule", ctx.sender, blockLabel.Regex)
+			log.Infof("Operator %s is trusted by the %s rule.", ctx.sender, blockLabel.Regex)
 			continue
 		}
 
 		// Undo the illegal operation.
 		if ctx.action == LabeledAction {
+			// Remove the label added illegally.
 			err := gc.RemoveLabel(owner, repo, ctx.number, ctx.label)
 
 			if err == nil {
-				log.Infof("Remove %s label added illegally", ctx.label)
+				log.Infof("Remove %s label added illegally.", ctx.label)
 			} else {
 				return fmt.Errorf("failed to remove illegal label added illegally, %s", err)
 			}
 		} else if ctx.action == UnlabeledAction {
+			// Restore the label removed illegally.
 			err := gc.AddLabel(owner, repo, ctx.number, ctx.label)
 
 			if err == nil {
-				log.Infof("Restore %s label removed illegally", ctx.label)
+				log.Infof("Restore %s label removed illegally.", ctx.label)
 			} else {
 				return fmt.Errorf("failed to restore the illegally removed label, %s", err)
 			}
 		}
+
+		// Reply to a message explaining why robot do this.
+		if len(blockLabel.Message) != 0 {
+			reason := ctx.sender + " " + ctx.action + " a label named " + ctx.label +
```

Comment:
Better use fmt.Sprintf.

And I think change message to `In response to adding or removing label xxx.` is better.

You can determine whether to add or delete based on the action.